### PR TITLE
[#375] Handle missing aud claim in access token

### PR DESCRIPTION
**Pull request by ll4strw**
Reject access tokens without aud claim

### CODE REVIEW

File: core/src/openid.cpp
Comment (MartinFlores751) [inline core/src/openid.cpp:1]: We would need to add the new configuration variable into `core/src/main.cpp`, in the schema:
https://github.com/irods/irods_client_http_api/blob/3824a09215a4b9b7e934178f2a7aa77b9654aa87/core/src/main.cpp#L208-L260

Something like the following should be fine:
```diff
                                "nonstandard_id_token_secret": {{
                                    "type": "string"
                                }},
+                               "strict_introspection_endpoint_aud": {{
+                                   "type": "boolean"
+                               }},
```

After that, we need to add the documentation for the new option in the README:
https://github.com/irods/irods_client_http_api/blob/3824a09215a4b9b7e934178f2a7aa77b9654aa87/README.md?plain=1#L253-L264

**Review by MartinFlores751 — CHANGES_REQUESTED** [review]
The change is very nice and simple. I like it.

Based on the discussions in the issue, we should have this be optional.

I'm suggesting we have a new configuration option `strict_introspection_endpoint_aud` to opt in to the strict check.

The name can change, of course. Open to suggestions.

**Comment by korydraughn** [comment]
@MartinFlores751 You can also make these edits and push to @ll4strw's branch.

**Comment by korydraughn** [comment]
And if you can't do that or feel comfortable doing that, you can create a branch from @ll4strw's, edit it, and then open another PR.

Either approach is fine, but whatever you do, make sure to leave a comment here so the story is clear to everyone.

**Comment by MartinFlores751** [comment]
Gotcha, I'll create a branch and open it in another PR.

File: core/src/openid.cpp [hunk 1]
```diff
@@ -154,6 +154,12 @@ namespace irods::http::openid
 				logging::warn("{}: Could not find our [client_id] in [aud]. Validation failed.", __func__);
 				return std::nullopt;
 			}
+		// Some IAM servers (e.g. keycloak) could be set up
+		// to exclude `aud' from a bearer token payload 	
+		// If no 'aud' was found in bearer token, do not accept 
+		} else {
```
[the remark this answers is not in the document]
Comment (MartinFlores751) [reply core/src/openid.cpp:160]: Handled in #378.

**Comment by korydraughn** [comment]
Once @MartinFlores751 resolves the review comments, we'll give the approval for merging.

File: core/src/openid.cpp
Comment (MartinFlores751) [reply core/src/openid.cpp:1]: Handled in #378.

File: core/src/openid.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (MartinFlores751) [reply core/src/openid.cpp:160]: This may be resolved.

File: core/src/openid.cpp
Comment (MartinFlores751) [reply core/src/openid.cpp:1]: This may be resolved.

**Comment by korydraughn** [comment]
Let's hold on merging this until #378 is complete.

**Comment by korydraughn** [comment]
#378 is now complete.

Merging.

**Review by korydraughn — APPROVED** [review]

**Pull request merged**